single_kernel_mongo/events/cluster.py

```diff
@@ -0,0 +1,156 @@
+        )
+        self.framework.observe(
+            self.charm.on[self.relation_name].relation_broken, self._on_relation_broken_event
+        )
+
+    def _on_database_requested(self, event: DatabaseRequestedEvent):
+        """Relation joined events."""
+        try:
+            self.manager.share_secret_to_mongos(event.relation)
+        except DeferrableFailedHookChecksError as e:
+            logger.info("Skipping database requested event: hook checks did not pass.")
+            defer_event_with_info_log(logger, event, str(type(event)), str(e))
+        except NonDeferrableFailedHookChecksError as e:
+            logger.info(f"Skipping {str(type(event))}: {str(e)}")
+
+    def _on_relation_event(self, event: RelationChangedEvent):
+        """Handle relation changed and relation broken events."""
+        try:
+            self.manager.update_keyfile_and_hosts_on_mongos(event.relation)
+        except DeferrableFailedHookChecksError as e:
+            defer_event_with_info_log(logger, event, str(type(event)), str(e))
+        except NonDeferrableFailedHookChecksError as e:
+            logger.info(f"Skipping {str(type(event))}: {str(e)}")
+
+    def _on_relation_broken_event(self, event: RelationBrokenEvent):
+        try:
+            self.manager.on_relation_broken(event.relation)
+        except DeferrableFailedHookChecksError as e:
+            defer_event_with_info_log(logger, event, str(type(event)), str(e))
+        except NonDeferrableFailedHookChecksError as e:
+            logger.info(f"Skipping {str(type(event))}: {str(e)}")
+
+
+class ClusterMongosEventHandler(Object):
+    """Event Handler for managing config server side events."""
+
+    def __init__(self, dependent: MongosOperator):
+        self.dependent = dependent
+        self.charm = self.dependent.charm
+        self.manager = self.dependent.cluster_manager
+        self.relation_name = self.manager.relation_name
+        super().__init__(parent=self.manager, key=dependent.cluster_manager.relation_name)
+
+        self.database_requirer_events = DatabaseRequirerEventHandlers(
+            self.charm, self.manager.data_interface
+        )
+
+        self.framework.observe(
+            self.charm.on[self.relation_name].relation_created,
+            self._on_relation_created,
+        )
+        self.framework.observe(
+            self.database_requirer_events.on.database_created, self._on_database_created
+        )
+        self.framework.observe(
+            self.charm.on[self.relation_name].relation_changed, self._on_relation_changed
+        )
+        self.framework.observe(
+            self.charm.on[self.relation_name].relation_departed,
+            self.dependent.check_relation_broken_or_scale_down,
+        )
+        self.framework.observe(
+            self.charm.on[self.relation_name].relation_broken, self._on_relation_broken
+        )
+
+    def _on_relation_created(self, event: RelationCreatedEvent):
+        self.manager.relation_created()
```

Comment:
similar comment on the naming from the manager standpoint and rest of occurrences